Sources/Brave/Frontend/Browser/New Tab Page/Sections/BraveNewsSectionProvider.swift

```diff
@@ -335,6 +338,14 @@ class BraveNewsSectionProvider: NSObject, NTPObservableSectionProvider {
       cell.content.smallHeadelineCardViews.right.feedView.setupWithItem(pair.second)
       cell.content.actionHandler = handler(from: { $0 == 0 ? pair.first : pair.second }, card: card, indexPath: indexPath)
       cell.content.contextMenu = contextMenu(from: { $0 == 0 ? pair.first : pair.second }, card: card, indexPath: indexPath)
+      // The rating card is presented
+      Preferences.Review.newsCardShownDate.value = Date()
```

Comment:
Technically this is also not guaranteed that the cell is visible anymore 😅 You may need to use `willDisplayCell` instead to guarantee it

Reply:
Also this is on the wrong switch case. Its on `headlinePair` instead of `headlineRatingCardPair` below this

Reply:
Thanks for the tip it is is moved to `willDisplayCell` Fix at https://github.com/brave/brave-ios/pull/7589/commits/ebbe26fa4af9e4be47f82ac8e078025b0f0f5031